src/python/pants/bsp/util_rules/targets.py

```diff
@@ -87,27 +87,28 @@ class BSPBuildTargets:
 
 @dataclass(frozen=True)
 class _ParseOneBSPMappingRequest:
+    name: str
     raw_specs: tuple[str, ...]
 
 
 @rule
-async def parse_one_bsp_mapping(request: _ParseOneBSPMappingRequest) -> Specs:
+async def parse_one_bsp_mapping(request: _ParseOneBSPMappingRequest) -> BSPBuildTargetInternal:
```

Comment:
Likewise, this seems like it could be a function instead.

Reply:
I may want to do some validation here which may require doing something with the `specs`. Although not clear to me whether I yet actually want to do that.